### PR TITLE
#160814322 Enable user login

### PR DESCRIPTION
**Pull request by cmplx-xyttmt**
#### What does this PR do?
Implements login functionality

#### Description of Task to be completed?
- Ensure users can log into the application.
- Ensure users can view relevant error messages when they enter invalid data

#### How should this be manually tested?
Run `npm start` and navigate to the login page enter some data

#### Any background context you want to provide?
Not applicable

#### What are the relevant pivotal tracker stories?
[#160814322](https://www.pivotaltracker.com/story/show/160814322)

#### Screenshots (if appropriate)
screen shot 2018-10-08 at 16 56 43
screen shot 2018-10-08 at 16 58 58

#### Questions:
Not applicable

### CODE REVIEW

**Comment by byarustev** [comment]
Thanks, Isaac for the good work. however, I am suggesting the button to be given some margin from the top such that there is some space between it and the note message. 
Then for the error messages, I think they would look better if they are made to start where the red line starts from.

**Comment by cmplx-xyttmt** [comment]
> Thanks, Isaac for the good work. however, I am suggesting the button to be given some margin from the top such that there is some space between it and the note message.
> Then for the error messages, I think they would look better if they are made to start where the red line starts from.

Thanks @byarustev for the review; I've fixed the issue with the button. As for the error messages, in my opinion, they are fine where they are.

**Comment by marthamareal** [comment]
Good work @cmplx-xyttmt , LGTM

**Comment by ruganda** [comment]
LGTM

File: src/actions/index.js [hunk 1]
```diff
@@ -1 +1,53 @@
...
+
+const Logout = () => {
+    return {
+        type: LOGOUT
+    }
+};
+
+export const handleLoginResponse = (user_data) => async (dispatch) => {
+
+    // console.log("User data: ", user_data);
```
Comment (inno-asiimwe) [inline src/actions/index.js]: Why do we have commented out code?

File: src/components/auth/Login.js [hunk 2]
```diff
@@ -1,13 +1,140 @@
 import React from 'react'
 
-const Login = () =>  {
+
+class LoginForm extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
```
Comment (inno-asiimwe) [inline src/components/auth/Login.js:8]: Does this component handle its own state?

Comment (cmplx-xyttmt) [reply src/components/auth/Login.js:8]: Yes. It handles only form input state. Would you prefer that I also handle that in global state?

File: src/actions/index.js (hunk 1 above)
Comment (cmplx-xyttmt) [reply src/actions/index.js]: Thanks for pointing that out, I've removed it

File: src/components/auth/Login.js (hunk 2 above)
Comment (marthamareal) [reply src/components/auth/Login.js:8]: I think its fine there.

Comment (inno-asiimwe) [reply src/components/auth/Login.js:8]: I think since we are using redux, let redux handle all the state.

Comment (cmplx-xyttmt) [reply src/components/auth/Login.js:8]: So, I should create actions that are dispatched every time a user adds a character in the input fields?

**Pull request merged**